src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.BoolValueSetFactory.cs

```diff
@@ -49,8 +49,7 @@ public IValueSet<bool> Related(BinaryOperatorKind relation, bool value)
 
             IValueSet IValueSetFactory.Related(BinaryOperatorKind relation, ConstantValue value)
             {
-                Debug.Assert(value.IsBoolean);
-                return Related(relation, value.BooleanValue);
+                return value.IsBad ? BoolValueSet.AllValues : Related(relation, value.BooleanValue);
```

Comment:
>value.IsBad [](start = 23, length = 11)

Are there other implementations of `IValueSetFactory.Related()` that need similar checks?

Reply:
All the other ones have the check.